File: packages/modules/document-picker/android/DocumentPicker.kt
// Copyright 2026 The Lynxpo Authors. All rights reserved.
// Licensed under the Apache License Version 2.0 that can be found in the
// LICENSE file in the root directory of this source tree.
package com.lynx.explorer.modules

import com.lynx.jsbridge.LynxMethod
import com.lynx.jsbridge.LynxModule
import android.content.Context

/**
 * Android counterpart of the iOS DocumentPicker. Exposes functionality to JS via
 * NativeModules.DocumentPicker, faithfully porting Expo's native method surface.
 */
class DocumentPicker(context: Context) : LynxModule(context) {
  @LynxMethod
  fun getDocumentAsync(options: String): Map<String, Any> {
    throw NotImplementedError()
  }

  @LynxMethod
  fun isAvailableAsync(): Boolean {
    throw NotImplementedError()
  }

}

File: packages/modules/document-picker/ios/DocumentPicker.m
// Copyright 2026 The Lynxpo Authors. All rights reserved.
// Licensed under the Apache License Version 2.0 that can be found in the
// LICENSE file in the root directory of this source tree.
#import <Foundation/Foundation.h>
#import <LynxModule/LynxModule.h>

@interface DocumentPicker() <LynxModule>
@end

@implementation DocumentPicker

+ (NSString *)name {
  return @"DocumentPicker";
}

- (NSDictionary<NSString *, NSString *> *)methodLookup {
  return @{
    @"documentAsync": @"documentAsyncWithOptions:",
    @"isAvailableAsync": @"isAvailableAsync:",
  };
}

- (NSDictionary *)documentAsyncWithOptions: WithOptions:(NSString *)options {
  return @{};
}

- (BOOL)isAvailableAsync: {
  return NO;
}

@end
